From cdef11b12d6feabf76faa97df3f0198c0b6f5bcd Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Fri, 29 May 2015 12:02:35 +0000
Subject: [PATCH 0590/1000] MANTIS 0035476 bugfix: Trace: wrong timestamp on
 "start new file"

trace.dll, tracehdl.dll, traceNoConfig.dll, tracehdlNoConfig.dll
V1.0.0.7

- use current timestamp for tracing in start new file
- start new trace file at 22:30 and 00:00
- fixed code formatting

git-svn-id: svn://localhost/SelfServiceCommon/trunk@863 90b65887-3827-0410-9a23-83215b262276
---
 SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp b/SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp
index ac4df3c6..1f885030 100644
--- a/SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp
+++ b/SelfServiceCommon/Massai/cpp/Trace/src/TraceCmdQueue.cpp
@@ -166,7 +166,9 @@ namespace MASSAI
     {
       TraceCmd* result= 0;
       mSysTime::Timestamp now= mSysTime::getTimestamp();
-      if(!sameDay(m_currentday + NEW_DAY_DELAY, now + NEW_DAY_DELAY))
+      // restart file at 00:00 and 22:30
+      if(!sameDay(m_currentday, now)
+        || !sameDay(m_currentday + NEW_DAY_DELAY, now + NEW_DAY_DELAY))
       {
         result= new TraceCmd(TRC_NEWDAY, now);
         m_currentday= now;
-- 
2.41.0.windows.1

